### ISSUE TITLE
[feature][grpc] Experiment gRPC gateway 

### ISSUE THREAD

**Issue by lampajr** [opendatahub-io/model-registry-bf4-kf#32]
**Is your feature request related to a problem? Please describe.**
UI can't directly do gRPC requests, therefore we should provide a REST proxy  service that can translates REST reqeusts to gRPC ones.

**Describe the solution you'd like**
One possible solution could be to autogenerate what is called `gRPC gateway` [1].

**Describe alternatives you've considered**
None

**Additional context**
This step should be automated in the build Makefile.

[1] https://grpc-ecosystem.github.io/grpc-gateway/

**Comment by lampajr** [comment]
Was able to setup a gRPC gateway based on the proto definition following these [instructions](https://github.com/grpc-ecosystem/grpc-gateway#usage).

Summaryizing what I did:
1. Using `protoc` plugin I automatically generated the proxy server (i.e., the gRPC gateway)
2. Create a new `gateway` command aiming to startup the gRPC gateway
3. Startup a dedicated server which setups a Dial connection with the existing gRPC server
4. Added a python script that, using REST requests, creates ArtifactType and Artifact instances.

Some considerations:
* Before starting up the gateway, the gRPC server must be up and running.
* The `protoc` is able to generate `Swagger` api specification, i.e., openapi v2 compatible. See https://github.com/grpc-ecosystem/grpc-gateway/issues/441

**Comment by rareddy** [comment]
Can we close this as we proved we can do this? We probably need another issue to include this gateway along with the REST API Server #59

**Comment by lampajr** [comment]
+1 I agree on closing this as the experimentation proved its feasibility.

>We probably need another issue to include this gateway along with the REST API Server https://github.com/opendatahub-io/model-registry/issues/59

Not sure this is needed right now as IIUC we do not need to expose the full mlmd gRPC interface as REST

**Comment by rareddy** [comment]
> Not sure this is needed right now as IIUC we do not need to expose the full mlmd gRPC interface as REST

I heard the envoy-generated gateway is used for training and experimentation, so we may need this enabled. But we could choose to come back to this as an addition over the REST API Model Registry going to support.
